packages/kbn-reporting/public/job_completion_notifications.ts

```diff
@@ -9,29 +9,55 @@
 import { JOB_COMPLETION_NOTIFICATIONS_SESSION_KEY } from '@kbn/reporting-common';
 import { JobId } from '@kbn/reporting-common/types';
 
-const set = (jobs: string[]) => {
-  sessionStorage.setItem(JOB_COMPLETION_NOTIFICATIONS_SESSION_KEY, JSON.stringify(jobs));
-};
+// Reading and writing to the local storage must be atomic,
+// i.e. performed in a single operation. This storage queue
+// Operations on the localStorage key can happen from various
+// parts of code. Using a queue to manage async operations allows
+// operations to process one at a time
+let operationQueue = Promise.resolve();
```

Comment:
Each plugin that imports the "kbn-reporting" package would receive its copy, hence different queues. Could this be a problem?

Reply:
Good point - it probably would.

Created d77186216a8